File: result/translate/starchat/python2java/java/FIND_SUBARRAY_WITH_GIVEN_SUM_1.java
public static int f_gold ( int arr [ ], int n, int sum ) {
    int curr_sum = arr [ 0 ];
    int start = 0;
    for ( int i = 1 ; i <= n ; i++ ) {
        while ( curr_sum > sum && start < i - 1 ) {
            curr_sum = curr_sum - arr [ start ];
            start++;
        }
        if ( curr_sum == sum ) {
            System.out.println ( "Sum found between indexes" );
            System.out.println ( start + " and " + ( i - 1 ) );
            return 1;
        }
        if ( i < n ) {
            curr_sum = curr_sum + arr [ i ];
        }
    }
    System.out.println ( "No subarray found" );
    return 0;
}

File: result/translate/chatgpt/java2python/python/FIND_SUBARRAY_WITH_GIVEN_SUM_1.py
def f_gold(arr, n, sum):
    curr_sum = arr[0]
    start = 0
    for i in range(1, n):
        while curr_sum > sum and start < i - 1:
            curr_sum -= arr[start]
            start += 1
        if curr_sum == sum:
            p = i - 1
            print("Sum found between indexes " + str(start) + " and " + str(p))
            return 1
        if i < n:
            curr_sum += arr[i]
    print("No subarray found")
    return 0